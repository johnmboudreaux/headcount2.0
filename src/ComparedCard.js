import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';

const ComparedCard = ({ comparedDataFunction, schoolsSelected, selectedCards }) => {
  // console.log(props);
  return (
    <div className="compare-container">

      {console.log(selectedCards(schoolsSelected))}
      {comparedDataFunction(selectedCards(schoolsSelected))}

    </div>
  );
};

ComparedCard.propTypes = {
  schoolsSelected: PropTypes.array,
  numberOfSelected: PropTypes.number
};

export default ComparedCard;
