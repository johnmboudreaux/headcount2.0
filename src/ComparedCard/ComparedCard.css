.compared-cards {
  color: #25ced1;
  height: 50px;
  text-align: center;
  padding-top: 15px; }
