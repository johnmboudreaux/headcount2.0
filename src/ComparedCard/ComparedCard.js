import React from 'react';
import PropTypes from 'prop-types';

const ComparedCard = ({ compareData, schoolsSelected, selectedCards }) => {
  return (
    <div className="compared-cards">
      {
        compareData(selectedCards(schoolsSelected)) &&
        compareData(selectedCards(schoolsSelected)).compared
      }
    </div>
  );
};

ComparedCard.propTypes = {
  selectedCards: PropTypes.func,
  compareData: PropTypes.func,
  schoolsSelected: PropTypes.array
};

export default ComparedCard;
