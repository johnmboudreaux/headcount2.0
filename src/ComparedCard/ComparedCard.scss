@import '../Styles/variables.scss';

.compared-cards {
  color: $pop-accent;
  height: 50px;
  text-align: center;
  padding-top: 15px;
}
