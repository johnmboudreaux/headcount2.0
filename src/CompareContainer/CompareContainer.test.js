import ReactDom from 'react-dom';
import React from 'react';
import { shallow, mount } from 'enzyme';
import ComparedCard from './ComparedCard';

describe('Compare Container', ()=>{
  it('should render compared data between data points', () => {
  //   const renderedComponent = shallow(
  //     <ComparedCard compareData={.5}/>
  //   );
  //   console.log(renderedComponent.debug());
  });
});
