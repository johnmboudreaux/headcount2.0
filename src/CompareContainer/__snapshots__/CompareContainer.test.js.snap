exports[`Compare Container should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    compareData={[Function]}
    handleCompareClick={[Function]}
    selectedCards={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="compare-container">
          <Card
                    dataNode={undefined}
                    handleClick={[Function]}
                    location={undefined}
                    numberOfSelected={undefined}
                    schoolsSelected={undefined} />
          <Card
                    dataNode={undefined}
                    handleClick={[Function]}
                    location={undefined}
                    numberOfSelected={undefined}
                    schoolsSelected={undefined} />
</div>,
        <div>
          <ComparedCard
                    compareData={[Function]}
                    schoolsSelected={undefined}
                    selectedCards={[Function]} />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              dataNode={undefined}
              handleClick={[Function]}
              location={undefined}
              numberOfSelected={undefined}
              schoolsSelected={undefined} />,
            <Card
              dataNode={undefined}
              handleClick={[Function]}
              location={undefined}
              numberOfSelected={undefined}
              schoolsSelected={undefined} />,
          ],
          "className": "compare-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "dataNode": undefined,
              "handleClick": [Function],
              "location": undefined,
              "numberOfSelected": undefined,
              "schoolsSelected": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "dataNode": undefined,
              "handleClick": [Function],
              "location": undefined,
              "numberOfSelected": undefined,
              "schoolsSelected": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <ComparedCard
            compareData={[Function]}
            schoolsSelected={undefined}
            selectedCards={[Function]} />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "compareData": [Function],
            "schoolsSelected": undefined,
            "selectedCards": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="compare-container">
            <Card
                        dataNode={undefined}
                        handleClick={[Function]}
                        location={undefined}
                        numberOfSelected={undefined}
                        schoolsSelected={undefined} />
            <Card
                        dataNode={undefined}
                        handleClick={[Function]}
                        location={undefined}
                        numberOfSelected={undefined}
                        schoolsSelected={undefined} />
</div>,
          <div>
            <ComparedCard
                        compareData={[Function]}
                        schoolsSelected={undefined}
                        selectedCards={[Function]} />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                dataNode={undefined}
                handleClick={[Function]}
                location={undefined}
                numberOfSelected={undefined}
                schoolsSelected={undefined} />,
              <Card
                dataNode={undefined}
                handleClick={[Function]}
                location={undefined}
                numberOfSelected={undefined}
                schoolsSelected={undefined} />,
            ],
            "className": "compare-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "dataNode": undefined,
                "handleClick": [Function],
                "location": undefined,
                "numberOfSelected": undefined,
                "schoolsSelected": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "dataNode": undefined,
                "handleClick": [Function],
                "location": undefined,
                "numberOfSelected": undefined,
                "schoolsSelected": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <ComparedCard
              compareData={[Function]}
              schoolsSelected={undefined}
              selectedCards={[Function]} />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "compareData": [Function],
              "schoolsSelected": undefined,
              "selectedCards": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
